.github/workflows/deploy.yml

```diff
@@ -1,19 +1,21 @@
 name: Build & Deploy
 on:
   push:
-    branches: [main]
+    branches: [21-set-up-cicd-pipeline-with-do] #Needs to be changed to release when on main.
 jobs:
   deploy:
     runs-on: ubuntu-latest
     steps:
     - name: Deploy Node Monitor
-      uses: appleboy/ssh-action@v0.1.2
+      uses: appleboy/ssh-action@v1.0.0
       with:
         host: ${{secrets.SSH_HOST}} # IP address of the server you wish to ssh into
         key: ${{secrets.SSH_KEY}} # Private or public key of the server
         username: ${{ secrets.SSH_USERNAME }} # User of the server you want to ssh into
-        script: |
-          mkdir test
-          cd test
-          git clone git@github.com:aviate-labs/node-monitor.git
+        script: | #change the script below to do 'git pull' in the correct directory in the droplet
+          cd test 
+          cd node-monitor
+          docker compose down
+          git pull
+          docker compose up --build -d
```

Comment:
I think this will leave whats called dangling docker images (docker doesnt delete old images when you rebuild)

You may have to call `docker image prune -a`
Otherwise we'll just be eating up space with new docker images

Reply:
Thanks for the suggestion! Although this leads to a longer duration to deploy. You can see the logs [here](https://github.com/aviate-labs/node-monitor/actions/runs/5713260213/job/15478330065). Although I don't think it is such a big issue